### PR TITLE
Remaining memory management fixes of RT #3198

### PR DESCRIPTION
**Pull request by mattcaswell**
This is a replacement pull request for #511.

Updated to take account of outstanding comments.

### CODE REVIEW

File: crypto/engine/eng_cryptodev.c [hunk 1]
```diff
@@ -1258,7 +1258,7 @@ static void zapparams(struct crypt_kop *kop)
 
     for (i = 0; i < kop->crk_iparams + kop->crk_oparams; i++) {
         if (kop->crk_param[i].crp_p)
```
Comment (richsalz) [inline crypto/engine/eng_cryptodev.c]: don't need the if check.

**Comment by mattcaswell** [comment]
If check removed....I'm confused. You added the reviewed label before I fixed it? Am I ok to push this?

**Comment by richsalz** [comment]
As with gitlab reviews, "plus one assuming you fix the issue"

**Comment by mattcaswell** [comment]
@richsalz can you review #1162 as well? I'd like to push them at the same time.

**Comment by richsalz** [comment]
I did.

**Comment by mattcaswell** [comment]
Pushed. Thanks

**Pull request closed without merging**